app/importers/clients/sftp_client.rb

```diff
@@ -39,20 +41,52 @@ def dir_entries(*args)
   # This returns an object that will reveal secure data if printed
   def sftp_session
     raise "SFTP information missing" unless sftp_info_present?
+
     if @sftp_session.nil?
-      @sftp_session = Net::SFTP.start(host, user, auth_mechanism)
+      start_sftp_session
     end
+
     @sftp_session
   end
 
+  def start_sftp_session
+    if ENV.fetch('USE_QUOTAGUARD', false)
+      start_sftp_session_with_proxy
+    else
+      start_sftp_session_no_proxy
+    end
+  end
+
+  def start_sftp_session_with_proxy
+    quotaguard = URI(ENV.fetch("QUOTAGUARDSTATIC_URL"))
+
+    proxy = Net::SSH::Proxy::HTTP.new(
+      quotaguard.host, quotaguard.port,
+      user: quotaguard.user, password: quotaguard.password
+    )
```

Comment:
This makes an HTTP connection with the quotaguard username and password.  Should this be HTTPS too?